### DISCUSSION TITLE
feedback - 7b models crashing

### DISCUSSION THREAD

**Discussion by Ar57m** [Mobile-Artificial-Intelligence/maid#28] [General]
hey, I am running your app on a Galaxy M52, 3B models runs fine on it, but 7b as pygmalion2 7b(3ks) and mystral pygmalion 7b(3km) it crashes and closes the app, on llama.cpp through termux it runs around 0.1~0.2 tokens/second.

**Comment by sfiannaca** [comment]
In my test, this bug occurs when the model surpasses the hardware capacity (out of memory). 

It seems that Termux utilizes less RAM compared to Flutter apps. 

You can attempt to utilize Q2_K, but the token speed of the 7B model is still relatively slow on medium-spec smartphone

**Reply by Ar57m** [reply]
thanks for the response and for the app.

Suggestion: if you can add more parameters like temperature, repetition penalty etc. 

I've notice that sometimes after some conversations, some messages are fully generated and then disappear(become empty).
And sometimes I close the app and enter again on it, it doesn't want to generate the response or crashes.

**Reply by Ar57m** [reply]
forgot to mention I'm running the version 1.0.3 on android.

**Reply by sfiannaca** [reply]
The app is currently under development, and I'm just a user like you, not the author

**Reply by danemadsen** [reply]
Try the new build, Ive exposed both those parameters.

**Comment by danemadsen** [comment]
Yeah certain 7B models crash the app because they use too much memory. In my testing if you're going to attempt to use 7B models you should try to use smaller Quant's.

**Reply by Ar57m** [reply]
I've noticed when I switch models, the app cache accumulate the models unnecessary.

**Reply by danemadsen** [reply]
Ill look into it

**Comment by Ar57m** [comment]
thanks for the update, the new release(maid 1.0.5 apk) it's very stable now, I can run pygmalion2 7b 3ks slowly and without any crashes, thanks again

**Reply by danemadsen** [reply]
That's great. Let me know if there's anything else I can do to improve the app.

**Reply by Ar57m** [reply]
The app is awesome now(1.0.8) without the giant cache and no more crashes, good work guys. 👏

Some suggestions: Support for llava models on mobile, min-p(parameter), some presets(character tab) for chatml, llama2, vicuna, alpaca prompt formating and some options to customize the background picture, AI 's picture, user's picture(like the aesthetic ui from koboldcpp).

**Comment by Ar57m** [comment]
Vali is developing a front-end like ST for android  https://github.com/Vali-98/ChatterUI

**Reply by danemadsen** [reply]
Good luck to him. He's using react native though and I'm using flutter. Different workflows.

**Comment by Ar57m** [comment]
can you please update the llama.cpp of the app to support stablelm models?

**Reply by Ar57m** [reply]
I've notice you have updated llama, but tried to run a stablelm3b model on the 1.1.2 version and is crashing

**Comment by danemadsen** [comment]
You need to use the latest actions release

https://github.com/MaidFoundation/Maid/actions/workflows/main.yml

Until I do another fixed point release.

With regards,
Dane Madsen
________________________________
